test/NuGet.Core.Tests/NuGet.Protocol.Tests/Providers/ReportAbuseResourceV3ProviderTests.cs

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using System.Threading;
+using Moq;
+using Newtonsoft.Json.Linq;
+using NuGet.Configuration;
+using NuGet.Protocol.Core.Types;
+using NuGet.Versioning;
+using Xunit;
+
+namespace NuGet.Protocol.Tests.Providers
+{
+    public class ReportAbuseResourceV3ProviderTests
+    {
+        private const string ResourceType = "ReportAbuseUriTemplate/3.0.0";
+
+        private readonly PackageSource _packageSource;
+        private readonly ReportAbuseResourceV3Provider _target;
+
+        public ReportAbuseResourceV3ProviderTests()
+        {
+            _packageSource = new PackageSource("https://unit.test");
+            _target = new ReportAbuseResourceV3Provider();
+        }
+
+        [Fact]
+        public async Task TryCreate_WhenResourceDoesNotExist_ReturnsFallbackUri()
+        {
+            var resourceProviders = new ResourceProvider[]
+            {
+                CreateServiceIndexResourceV3Provider(),
+                _target
+            };
+            var sourceRepository = new SourceRepository(_packageSource, resourceProviders);
+
+            var result = await _target.TryCreate(sourceRepository, CancellationToken.None);
+
+            Assert.True(result.Item1);
+            Assert.IsType<ReportAbuseResourceV3>(result.Item2);
+            Assert.Equal("https://www.nuget.org/packages/MyPackage/1.0.0/ReportAbuse",
+                         ((ReportAbuseResourceV3)result.Item2).GetReportAbuseUrl("MyPackage", NuGetVersion.Parse("1.0.0")).OriginalString);
+        }
+
+        [Theory]
+        [InlineData("https:\\bar")]
+        [InlineData("foo")]
+        [InlineData("../somepath")]
+        [InlineData(null)]
+        [InlineData("")]
+        [InlineData("  \t\n")]
+        public async Task TryCreate_WhenResourceHasInvalidAbsoluteUri_ReturnsFallbackUri(string uri)
+        {
+            var serviceEntry = new RawServiceIndexEntry(uri, ResourceType);
+            var resourceProviders = new ResourceProvider[]
+            {
+                CreateServiceIndexResourceV3Provider(serviceEntry),
+                _target
+            };
+            var sourceRepository = new SourceRepository(_packageSource, resourceProviders);
+
+            var result = await _target.TryCreate(sourceRepository, CancellationToken.None);
+
+            Assert.True(result.Item1);
+            Assert.IsType<ReportAbuseResourceV3>(result.Item2);
+            Assert.Equal("https://www.nuget.org/packages/MyPackage/1.0.0/ReportAbuse",
+                         ((ReportAbuseResourceV3)result.Item2).GetReportAbuseUrl("MyPackage", NuGetVersion.Parse("1.0.0")).OriginalString);
+        }
+
+        [Fact]
```

Comment:
If the behavior is incorrect, can we test the correct behavior and skip the test and link the issue in the skip comment?

```suggestion
        [Fact(Skip = "https://github.com/nuget/home/issues/****")]
```